cupy/core/include/cupy/carray.cuh

```diff
@@ -228,32 +324,19 @@ public:
       // contiguous arrays can be directly addressed by the
       // numeric value, avoiding expensive 64 bit operations in cuda
       return data_[idx];
-    }
-    // 64-bit mults and divs are pretty expensive and can 
-    // lead to severe perforamance degradation in computation bound
-    // kernels
-    const char* ptr = reinterpret_cast<const char*>(data_);
-    if (use_32bit_indexing) {
-      int i = static_cast<int>(idx);
-      for (int dim = ndim; --dim > 0; ) {
-        int shape_dim = static_cast<int>(shape_[dim]);
-        ptr += static_cast<int>(strides_[dim]) * (i % shape_dim);
-        i /= shape_dim;
-      }
-      if (ndim > 0) {
-        ptr += static_cast<int>(strides_[0]) * i;
-      }
     } else {
-      ptrdiff_t i = idx;
+      // 64-bit mults and divs are pretty expensive and can lead to severe
+      // performance degradation in computation bound kernels
+      const char* ptr = reinterpret_cast<const char*>(data_);
+      index_t i = static_cast<index_t>(idx);
       for (int dim = ndim; --dim > 0; ) {
-        ptr += static_cast<ptrdiff_t>(strides_[dim]) * (i % shape_[dim]);
-        i /= shape_[dim];
-      }
-      if (ndim > 0) {
-        ptr += static_cast<ptrdiff_t>(strides_[0]) * i;
+          index_t shape_dim = static_cast<index_t>(shape_[dim]);
+          ptr += static_cast<index_t>(strides_[dim]) * (i % shape_dim);
+          i /= shape_dim;
       }
+      ptr += static_cast<index_t>(strides_[0]) * i;
```

Comment:
Question: We don't need to check if `ndim>0` here or not because in that case `strides_[0]` is 0?